packages/contracts/deploy/deployContracts.ts

```diff
@@ -37,6 +37,12 @@ export const deployGreenproof = async (options: InitContractOptions) => {
     claimManagerAddress = VOLTA_CLAIM_MANAGER,
     claimRevokerAddress = VOLTA_CLAIM_REVOKER,
     roles = {},
+    batchConfig = {
```

Comment:
Not sure how you tested that, but probably issuance gonna have the biggest payload, therefore it probably will have lowest batchIssuanceSize

Reply:
By the way Kevin, why contract cares about the batch size at all?

Reply:
> Not sure how you tested that, but probably issuance gonna have the biggest payload, therefore it probably will have lowest batchIssuanceSize. Issuance always carries the most metadata

Yes, that is what @JonaszJestem pointed also during the discussion we had. The maximum number of transaction in each type of batch will be configured during project setting / deployment, but it will be up to the backend to adjust the nb of transactions for each call. 
In the tests, I set all the batch limits to 20, but in production maybe we will have to set it to 10 for issuance and 20 for the other batches

Reply:
> By the way Kevin, why contract cares about the batch size at all if the backend will be managing the batches itself?

To prevent a wrong usage of this batching feature, it is better to enforce the limitation directly on the contract side. We may face situations where : 
- Some unadvised users may send a lot of Tx which will cause the execution to revert.
- Since the contract will be public, some could decide no to use the backend SDK and directly call the contract batch functions either from the explorer or from they own script.

Reply:
But what if a user can submit 100 tx of transfers, because they have very small payload? Then he cannot configure the contract, so he has to fork it. I don't know, personally I feel like this is not good argument, that Tx may revert. After all, if you submit 1000 tx of transfers, and hit limit in the contract currently implemented, your execution is reverted anyway, right? Just with a different error message. So the end result is exactly the same, but not putting the limit inside the contract at least gives you some space for experiments what batch size works for your particular application.

Also, as I said, this limit in contract doesn't solve a problem as well, because the number of allowed transactions is determined by the payload size, therefore the gas limit. It's not sheer number of operations, but the overall size of the batch in bytes. And checking that thing makes no sense, because it's checked by EVM anyway

Reply:
> But what if a user can submit 100 tx of transfers, because they have very small payload? Then he cannot configure the contract, so he has to fork it

Either the user knows that 100 tx may work before initialization, then the contract should be initialized with this value. Either this is not known before and we can add a setter function to adjust this size afterwards. 
But it's worth mentioning that we are not searching for the perfect number. We are providing a way to improve a bit the scalability. And at the end of the day, this number is limited by the gas limit. Which brings us to your second point : "**why setting this limitation if the Tx will always revert when we hit the gas limit ?**"

The simple answer is that we are providing a more controlled way of dealing with this limitation. If we don't provide a customized error for this situation, the revert message (if existing) may sometime be hard to relate to that  issue. Letting the EVM handle limitation error is affordable in most situations, but may introduce awful debug/audit experience. 
Another aspect is that when we hit the gas limit for a block, the transaction fails but the fees gets consumed and not returned; this which can lead to lost gas fees and a poor user experience. Thus, setting up safeguards which reverts sooner to prevent this could be a good idea.